src/fcio.c

```diff
@@ -763,23 +849,55 @@ static inline void fcio_get_sparseevent(FCIOStream stream, fcio_event *event, in
   event->timeoffset_size = FCIOReadInts(stream,10,event->timeoffset)/sizeof(int);
   event->timestamp_size = FCIOReadInts(stream,10,event->timestamp)/sizeof(int);
   event->deadregion_size = FCIOReadInts(stream,10,event->deadregion)/sizeof(int);
-  
+
   FCIOReadInts(stream,1,&event->num_traces);
   FCIOReadUShorts(stream,event->num_traces,event->trace_list);
-  int i; 
+  int i;
   for(i=0; i<event->num_traces; i++)
     FCIOReadUShorts(stream,tracesamples,&event->traces[event->trace_list[i]*tracesamples]);
 
-  if (debug > 3) 
+  if (debug > 3)
   {
-    int i; 
+    int i;
     fprintf(stderr,"FCIO/fcio_get_sparse_event: type %d pulser %g, offset %d %d %d ",event->type,event->pulser,event->timeoffset[0],event->timeoffset[1],event->timeoffset[2]);
     fprintf(stderr,"timestamp "); for (i = 0; i < 10; i++) fprintf(stderr,"%d ",event->timestamp[i]);
     fprintf(stderr,"dead "); for (i = 0; i < 10; i++) fprintf(stderr,"%d ",event->deadregion[i]);
-    //fprintf(stderr," traces "); 
-    //for (i = 0; i < event->num_traces; i++) fprintf(stderr," %d",event->trace_list[i]); 
+    //fprintf(stderr," traces ");
+    //for (i = 0; i < event->num_traces; i++) fprintf(stderr," %d",event->trace_list[i]);
     fprintf(stderr,"\n");
-    
+
+  }
+}
+
+static inline void fcio_get_eventheader(FCIOStream stream, fcio_config* config, fcio_event *event)
+{
+  FCIOReadInt(stream,event->type);
+  FCIOReadFloat(stream,event->pulser);
+  event->timeoffset_size = FCIOReadInts(stream,10,event->timeoffset)/sizeof(int);
+  event->timestamp_size = FCIOReadInts(stream,10,event->timestamp)/sizeof(int);
+  event->deadregion_size = FCIOReadInts(stream,10,event->deadregion)/sizeof(int);
+  FCIOReadInts(stream,1,&event->num_traces);
+  FCIOReadUShorts(stream,FCIOMaxChannels,event->trace_list);
```

Comment:
Need to check the return value here to safely iterate over the trace list below.